ui/src/main/js/page/user/user/UserDeleteModal.js

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState } from 'react';
+import { createUseStyles } from 'react-jss';
+import { useDispatch } from 'react-redux';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { deleteUser, fetchUsers } from 'store/actions/users';
+import Modal from 'common/component/modal/Modal';
+
+const useStyles = createUseStyles({
+    deleteConfirmMessage: {
+        margin: [0, 0, '20px', '30px'],
+        fontSize: '16px',
+        fontWeight: 'bold'
+    },
+    cardContainer: {
+        display: 'flex',
+        marginLeft: '50px'
+    },
+    userCard: {
+        display: 'flex',
+        border: ['1px', 'solid', '#D9D9D9'],
+        borderRadius: '5px',
+        backgroundColor: '#e8e6e6',
+        padding: '8px',
+        margin: [0, '50px', '10px', '20px'],
+        width: '250px'
+    },
+    userIcon: {
+        flexBasis: '20%',
+        backgroundColor: 'white',
+        border: ['1px', 'solid', '#D9D9D9'],
+        borderRadius: '5px',
+        height: '50px',
+        paddingTop: '5px',
+        textAlign: 'center'
+    },
+    userInfo: {
+        flexGrow: 1,
+        padding: ['5px', 0, 0, '15px']
+    }
+});
+
+
+const UserDeleteModal = ({ isOpen, toggleModal, data, selected }) => {
+    const classes = useStyles();
+    const dispatch = useDispatch();
+    const [selectedUsers, setSelectedUsers] = useState(getStagedForDelete());
+    const isMultiUserDelete = selectedUsers.length > 1;
+
+    function getStagedForDelete() {
+        const staged = data.filter(user => selected.includes(user.id));
+        return staged.map(user => ({ ...user, staged: true }));
+    }
+
+    useEffect(() => {
+        setSelectedUsers(getStagedForDelete());
```

Comment:
need to move `getStagedForDelete()` function to above the useEffect because you are using it before it is defined.